span.go

```diff
@@ -6,16 +6,13 @@ import (
 	"time"
 )
 
-// Span represents an active, un-finished span in the opentracing system.
+// Span represents an active, un-finished span in the OpenTracing system.
 //
-// Spans are created by the Tracer interface and Span.StartChild.
+// Spans are created by the Tracer interface.
 type Span interface {
 	// Sets or changes the operation name.
 	SetOperationName(operationName string) Span
 
-	// Creates and starts a child span.
-	StartChild(operationName string) Span
```

Comment:
Also this allows putting encoding and decoding next to each other on the
tracer.

On Wed, Feb 10, 2016, 00:48 bhs <notifications@github.com> wrote:

> In span.go
> <https://github.com/opentracing/opentracing-go/pull/50#discussion_r52417346>
> :
>
> >  type Span interface {
> >  	// Sets or changes the operation name.
> >  	SetOperationName(operationName string) Span
> >
> > -	// Creates and starts a child span.
> > -	StartChild(operationName string) Span
>
> This really doesn't need to be in Span as long as we are willing to have
> a Span.Tracer() accessor. Since I can't imagine a Span impl that doesn't
> have access to its Tracer (to handle Finish() if nothing else), I like
> this change. See the new StartChildSpan (global) helper method.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opentracing/opentracing-go/pull/50/files#r52417346>.
>